lib/elixir/lib/access.ex

```diff
@@ -281,21 +300,22 @@ defmodule Access do
   end
 
   @doc """
-  Gets and updates the given key in a container (a map, keyword
-  list, or struct that implements the `Access` behaviour).
+  Gets and updates the given key in a `container` (a map, a keyword list,
+  a struct that implements the `Access` behaviour).
 
   The `fun` argument receives the value of `key` (or `nil` if `key` is not
-  present) and must return a two-element tuple `{get_value, updated_term}`: the
-  "get" value `get_value` (the retrieved value, which can be operated on before
-  being returned) and the new value to be stored under `key`
-  (`updated_term`). `fun` may also return `:pop`, which means the current value
+  present in `container`) and must return a two-element tuple `{get_value, update_term}`:
+  the "get" value `get_value` (the retrieved value, which can be operated on before
+  being returned) and the new value to be stored under `key` (`update_value`).
+  `fun` may also return `:pop`, which means the current value
   should be removed from the container and returned.
 
   The returned value is a two-element tuple with the "get" value returned by
   `fun` and a new container with the updated value under `key`.
   """
-  @spec get_and_update(container :: t, key, (value -> {get_value, update_value} | :pop)) ::
-        {get_value, container :: t} when get_value: var, update_value: value
+  @spec get_and_update(container, key, (value | nil -> {get_value, update_value} | :pop)) ::
+        {get_value, new_container :: container} when container: keyword | struct | map,
```

Comment:
If we are `container: keyword | struct | map` in multiple places, I would say we need three types:

```
@type container :: keyword | struct | map
@type nil_container :: nil
@type any_container :: any
@type t :: container | any_container | nil_container
```